Lambdas/GHHooks/GHMiddleware.swift

```diff
@@ -57,15 +86,32 @@ actor GHMiddleware: ClientMiddleware {
 
         do {
             let response = try await next(request, baseURL)
-            
-            logger.debug("Got response from Github", metadata: [
+
+            logger.debug("Got response from GitHub", metadata: [
                 "response": "\(response.fullDescription)",
                 "requestID": .stringConvertible(requestID),
             ])
 
-            return response
+            /// If this is not _the_ retry,
+            /// and if the authorization is set to use an authenticator,
+            /// and if the response status is `401 Unauthorized`,
+            /// then retry the request with a force-refreshed token.
+            if !isRetry,
+               case .authenticator = authorization,
+               response.statusCode == 401 {
```

Comment:
When possible, we should yank in the new `HTTPTypes` package so you can write `.unauthorized` instead of `401` here...

Reply:
There are so many packages that can use the HTTPTypes package ... this is all from the OpenAPI-Generator so not exactly under my control but it's possible i do a PR for it someday.